io-engine/src/grpc/v1/stats.rs

```diff
@@ -216,6 +216,51 @@ impl StatsRpc for StatsService {
         .await
     }
 
+    async fn get_replica_io_stats(
+        &self,
+        request: Request<ListStatsOption>,
+    ) -> GrpcResult<ReplicaIoStatsResponse> {
+        self.shared(self.replica_svc.rw_lock().await, async move {
+            let args = request.into_inner();
+            let rx = rpc_submit::<_, _, CoreError>(async move {
+                let mut lvol = Vec::new();
+                if let Some(bdev) = UntypedBdev::bdev_first() {
+                    lvol = bdev
+                        .into_iter()
+                        .filter(|b| b.driver() == "lvol")
+                        .map(|b| Lvol::try_from(b).unwrap())
+                        .collect();
+                }
+                let replica_stats_future: Vec<_> = if let Some(name) = args.name
```

Comment:
We should first check if `name` is present in args and search for the specific lvol. If not, then take all lvol bdevs. Will save time.

Reply:
We dont have a way of directly getting a specific lvol. Like how we have lookup fn for Lvs and Nexus.

So we list all bdev, then filter of type lvol and get a specifc lvol if arg.name is present. 

Exporter will always list all replicas. We have gvien an option to specify name as it will be easier for us to filter a particular resource stat with io-engine-client.

Reply:
ok, but we can still optimise. If the args.name is provided and this lvol was the first registered bdev among say 1000s, then the current logic will first iterate over 1000, and then weed out 999 to keep this one. Instead if we run iteration to find this args.name with `.filter(|b| b.driver() == "lvol") && b.name() == args.name` , then we don't need to go all the way till 1000 always.

Reply:
Even if we filter with name also along with lvol . .`filter(|b| b.driver() == "lvol") && b.name() == args.name` . This doesnt break after getting a first match. It will proceed iterating to the end of elements right?

Reply:
There is I think similar  `find` or`find_map` that can be used. They short-circuit upon finding the matching value.

Reply:
Done